### PR TITLE
feat: bar chart

### PR DESCRIPTION
**Pull request by elacallefactorial**
## 🚪 Why?

Factorial one doesn't have a Bar component in order to use it with other charts components

## 🔑 What?

This PR adds a Bar chart to draw Bar components and Lines in the insights components and data graph component; this Bar chart has the following interface:

- **dataConfig** Configures each data point with a label and colour
- **xAxis** 
  - **hide** Hides the X-axis
  - **formatter** Arguments that receive a function to customize the X-axis labels
- **lines** Draw a line over the Bar components
- **data** required data by the Bar component to draw the Bars and Lines
- **yAxis** 
  - **hide** Hides the X-axis
  - **formatter** Arguments that receive a function to customize the X-axis labels

## 🏡 Context

https://www.figma.com/design/9pLsRzdinid0ha0qRWta2D/Employee-Profile?node-id=378-19759&t=cFvdNpMxrdkyqDqn-0

## 📸 Visual proof

Screenshot 2024-07-24 at 11 38 39

### CODE REVIEW

**Comment by dani-moreno** [comment]
Hey, good job @elacallefactorial, looking good!

If you take a look at the [recharts Bar API](https://recharts.org/en-US/api/Bar), you'll see that there are a few props that may be worth adding. Especially `layout`, and probably `label`.

File: lib/components/Charts/BarChart/index.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      },
+      mobile: {
+        label: "Mobile",
+        color: "hsl(var(--chart-2))",
+      },
+    },
+    xAxis: {
+      hide: false,
+    },
+    lines: true,
```
Comment (josepjaume) [inline lib/components/Charts/BarChart/index.stories.tsx:40]: Is that something we really need? From the screenshot, it looks a bit redundant.

File: lib/components/Charts/utils/bar.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { ChartItem } from "./types"
+
+export function prepareData<Keys extends string>(data: ChartItem<Keys>[]) {
+  return data.map((item) => ({ x: item.label, ...item.values }))
+}
```
Comment (josepjaume) [inline lib/components/Charts/utils/bar.ts:5]: Could we perhaps use this function on the `LineChart` component as well so it's not duplicated?

**Review by josepjaume — APPROVED** [review]
Requested a couple changes but preemptively approving.

**Comment by josepjaume** [comment]
Nice job! Could you maybe consider adding an equivalent `BarChartInsight` just like the rest? Should be a handful of lines. Also please remember to add it on their `exports.ts` file, as it's what we actually export (the rest stays private).

**Comment by elacallefactorial** [comment]
> recharts Bar API

Hi @dani-moreno I've created other PR with this feature https://github.com/factorialco/factorial-one/pull/341

**Pull request merged**